File: runner/runner.go
package runner

import (
	"fmt"
	"os"

	"github.com/kx0101/liakos-language/evaluator"
	"github.com/kx0101/liakos-language/lexer"
	"github.com/kx0101/liakos-language/object"
	"github.com/kx0101/liakos-language/parser"
)

func RunFile(filePath string) {
	content, err := os.ReadFile(filePath)
	if err != nil {
		fmt.Printf("Could not read file: %s\n", err)
		os.Exit(1)
	}

	l := lexer.New(string(content))
	p := parser.New(l)

	program := p.ParseProgram()
	if len(p.Errors()) != 0 {
		printParserErrors(p.Errors())
		os.Exit(1)
	}

	env := object.NewEnvironment()

	for _, stmt := range program.Statements {
		evaluated := evaluator.Eval(stmt, env)
		if evaluated != nil && evaluated.Type() != object.NULL_OBJ {
			fmt.Println(evaluated.Inspect())
		}
	}
}

func printParserErrors(errors []string) {
	fmt.Println("Parser errors:")

	for _, msg := range errors {
		fmt.Printf("\t%s\n", msg)
	}
}

File: src/runner.rs
use std::fs;
use std::process;

use crate::environment::Environment;
use crate::evaluator::eval_statement;
use crate::lexer::Lexer;
use crate::object::Object;
use crate::parser::Parser;

pub fn run_file(file_path: &str) {
    let content = match fs::read_to_string(file_path) {
        Ok(c) => c,
        Err(e) => {
            eprintln!("Could not read file: {}", e);
            process::exit(1);
        }
    };
    let lexer = Lexer::new(content);
    let mut parser = Parser::new(lexer);
    let program = parser.parse_program();
    if !parser.errors.is_empty() {
        print_parser_errors(&parser.errors);
        process::exit(1);
    }
    let env = Environment::new();
    for stmt in &program.statements {
        let evaluated = eval_statement(stmt, env.clone());
        if !matches!(evaluated, Object::Null) {
            println!("{}", evaluated.inspect());
        }
    }
}

fn print_parser_errors(errors: &[String]) {
    println!("Parser errors:");
    for msg in errors {
        println!("\t{}", msg);
    }
}
